base/message_loop/message_pump_io_starboard.cc

```diff
@@ -39,32 +39,30 @@ MessagePumpIOStarboard::SocketWatcher::~SocketWatcher() {
   }
 }
 
-bool MessagePumpIOStarboard::SocketWatcher::StopWatchingSocket() {
-  watcher_ = nullptr;
-  interests_ = kSbSocketWaiterInterestNone;
-  if (!SbSocketIsValid(socket_)) {
-    pump_ = nullptr;
-    // If this watcher is not watching anything, no-op and return success.
-    return true;
-  }
-
-  SbSocket socket = Release();
+bool MessagePumpIOStarboard::SocketWatcher::UnregisterInterest(int interests) {
   bool result = true;
-  if (SbSocketIsValid(socket)) {
-    DCHECK(pump_);
-#if defined(STARBOARD)
+  if (SbSocketIsValid(socket_)) {
     // This may get called multiple times from TCPSocketStarboard.
     if (pump_) {
-      result = pump_->StopWatching(socket);
-    }
-#else
-    result = pump_->StopWatching(socket);
-#endif
+      result = pump_->UnregisterInterest(socket_, interests, this);
+    } else
+      Release();
+  } else {
+    interests_ = 0;
+  }
+  if (!interests_) {
+    DCHECK(!SbSocketIsValid(socket_));
+    pump_ = nullptr;
+    watcher_ = nullptr;
   }
-  pump_ = nullptr;
   return result;
 }
 
+bool MessagePumpIOStarboard::SocketWatcher::StopWatchingSocket() {
+  return UnregisterInterest(kSbSocketWaiterInterestRead ||
```

Comment:
Shouldn't this be `kSbSocketWaiterInterestRead | kSbSocketWaiterInterestWrite`? (instead of `||`)

Reply:
Good catch.